src/apps/Altinn.AccessManagement/src/Altinn.AccessManagement.Core/Services/ConsentService.cs

```diff
@@ -0,0 +1,760 @@
+    {
+        private readonly IConsentRepository _consentRepository = consentRepository;
+        private readonly IPartiesClient _partiesClient = partiesClient;
+        private readonly ISingleRightsService _singleRightsService = singleRightsService;
+        private readonly IResourceRegistryClient _resourceRegistryClient = resourceRegistryClient;
+        private readonly IAMPartyService _ampartyService = ampartyService;
+        private readonly IMemoryCache _memoryCache = memoryCache;
+        private readonly IProfileClient _profileClient = profileClient;
+
+        private const string _consentRequestStatus = "Status";
+        private const string ResourceParam = "Resource";
+
+        /// <inheritdoc/>
+        public async Task<Result<ConsentRequestDetails>> CreateRequest(ConsentRequest consentRequest, ConsentPartyUrn performedByParty, CancellationToken cancellationToken)
+        {
+            Result<ConsentRequest> result = await ValidateAndSetInternalIdentifiers(consentRequest, cancellationToken);
+
+            if (result.IsProblem)
+            {
+                return result.Problem;
+            }
+
+            performedByParty = await MapFromExternalIdenity(performedByParty, cancellationToken);
+
+            ConsentRequestDetails requestDetails = await _consentRepository.CreateRequest(result.Value, performedByParty, cancellationToken);
+            requestDetails.From = consentRequest.From;
+            requestDetails.To = consentRequest.To;
+            foreach (ConsentRequestEvent consentRequestEvent in requestDetails.ConsentRequestEvents)
+            {
+                consentRequestEvent.PerformedBy = await MapToExternalIdenity(consentRequestEvent.PerformedBy, cancellationToken);
+            }
+
+            return requestDetails;
+        }
+
+        /// <inheritdoc/>
+        public async Task<Result<ConsentRequestDetails>> RejectRequest(Guid consentRequestId, Guid performedByParty, CancellationToken cancellationToken)
+        {
+            ValidationErrorBuilder errors = default;
+            ConsentRequestDetails details = await _consentRepository.GetRequest(consentRequestId, cancellationToken);
+            if (details.ConsentRequestStatus == ConsentRequestStatusType.Rejected)
+            {
+                await SetExternalIdentities(details, cancellationToken);
+                return details;
+            }
+
+            if (details.ConsentRequestStatus != ConsentRequestStatusType.Created)
+            {
+                errors.Add(ValidationErrors.ConsentCantBeRejected, _consentRequestStatus);
+            }
+
+            if (errors.TryBuild(out var beforeErrorREsult))
+            {
+                return beforeErrorREsult;
+            }
```

Comment:
This isn't validation logic. It should not be creating validation errors.

```suggestion
            if (details.ConsentRequestStatus != ConsentRequestStatusType.Created)
            {
                return Problems.ConsentCantBeRejected;
            }
```